app/src/main/java/org/wikipedia/talk/TalkTopicsActivity.kt

```diff
@@ -0,0 +1,211 @@
+package org.wikipedia.talk
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.disposables.CompositeDisposable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import kotlinx.android.synthetic.main.activity_talk_topics.*
+import org.wikipedia.Constants
+import org.wikipedia.R
+import org.wikipedia.WikipediaApp
+import org.wikipedia.activity.BaseActivity
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.okhttp.HttpStatusException
+import org.wikipedia.dataclient.page.TalkPage
+import org.wikipedia.settings.languages.WikipediaLanguagesActivity
+import org.wikipedia.settings.languages.WikipediaLanguagesFragment
+import org.wikipedia.util.L10nUtil
+import org.wikipedia.util.StringUtil
+import org.wikipedia.util.log.L
+import org.wikipedia.views.DrawableItemDecoration
+import org.wikipedia.views.FooterMarginItemDecoration
+
+class TalkTopicsActivity : BaseActivity() {
+    private var wikiSite: WikiSite = WikipediaApp.getInstance().wikiSite
+    private var userName: String = ""
+    private val disposables = CompositeDisposable()
+    private val topics = ArrayList<TalkPage.Topic>()
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_talk_topics)
+
+        if (intent.hasExtra(EXTRA_LANGUAGE)) {
+            wikiSite = WikiSite.forLanguageCode(intent.getStringExtra(EXTRA_LANGUAGE).orEmpty())
+        }
+        userName = intent.getStringExtra(EXTRA_USER_NAME).orEmpty()
+        title = getString(R.string.talk_user_title, StringUtil.removeUnderscores(userName))
+
+        talkRecyclerView.layoutManager = LinearLayoutManager(this)
+        talkRecyclerView.addItemDecoration(FooterMarginItemDecoration(0, 80))
+        talkRecyclerView.addItemDecoration(DrawableItemDecoration(this, R.attr.list_separator_drawable, drawStart = false, drawEnd = false))
+        talkRecyclerView.adapter = TalkTopicItemAdapter()
+
+        talkErrorView.setBackClickListener {
+            finish()
+        }
```

Comment:
Please also set up the `setRetryClickListener` in case the button does not work when losing an internet connection.

Reply:
ok!